src/main/frontend/src/code-quarkus/api/quarkus-project-utils.ts

```diff
@@ -94,6 +94,23 @@ const getParams = (paramName: string): string | null => {
   return searchParams.get(paramName);
 }
 
+export function syncParamsInQuery(filterParam: string = '', project: QuarkusProject | undefined): void {
+  if (!project) {
+    window.history.replaceState(null, '', `/?${formatParam(queryName, filterParam)}`);
+    return;
+  }
+
+  window.history.replaceState(null, '', `/?${generateProjectQuery(project)}&${formatParam(queryName, filterParam)}`)
```

Comment:
I think we need to:
- improve the `generateProjectQuery` so that it doesn't add unchanged/default options
- add a debounce (~ 1 or 2 seconds) to avoid changing it too much 
- avoid adding the `&` in the end when the filter is empty

Reply:
Cool! great points that you've looked!!

I'll look about it

Reply:
What I did in the new code..

* Improve `generateProjectQuery` , now it just add what is not default.
* Added a debounce for project updates and filter updates to avoid changing it too much.
* Fixing the query param to avoid the `&` in the end

:)